index.js

```diff
@@ -335,10 +338,11 @@
   var TypeRep = $.NullaryType(
     'sanctuary/TypeRep',
     function(x) {
-      return R.type(x) === 'Function' ||
-             (x != null &&
-              R.type(x.name) === 'String' &&
-              R.type(x.test) === 'Function');
+      return R.type(x) === 'Function'
+          && R.type(global_[x.name]) === 'Function'
+          || x != null
+          && x.prototype != null
+          && R.type(x.prototype['@@type']) === 'String';
```

Comment:
The three lines above handle user-defined type representatives, identified by the presence of a `@@type` property on the object's `prototype`.